src/arch/freeRTOS/forte_Init.h

```diff
@@ -36,17 +36,16 @@ extern "C" {
     FORTE_WRONG_ENDIANESS,
     FORTE_WRONG_PARAMETERS,
     FORTE_ARCHITECTURE_NOT_READY,
+    FORTE_COULD_NOT_CREATE_DEVICE
   };
 
-  typedef void* TForteInstance;
-
   /**
    * \brief Start forte instance
    * @param paPort The port on which to forte will listen. Use 0 for default (normally 61499)
    * @param paResultInstance Address of an instance of forte where the new instance is stored
    * @return FORTE_OK if no error occurred, other values otherwise
    */
-  FORTE_SHARED_PREFIX int FORTE_SHARED_CALL forteStartInstance(unsigned int paPort, TForteInstance* paResultInstance);
+  FORTE_SHARED_PREFIX int FORTE_SHARED_CALL forteStartInstance(unsigned int paPort);
```

Comment:
From what I remember, this methods were intended to be used as a `C` interface to the dynamic/static forte library linked to `C` code. 

The idea of having the forte pointer managed by the caller was to avoid problems from a global pointer in a dynamic library which would be shared by many callers. I understand that this is maybe not the case now and it's a very strange case, but thinking this file as a purely `C` interface I would keep the pointer management to the caller. 

Also, I'm not sure why we have so many copies of this file, since the idea of the interface is to have a unique one for all cases.

Reply:
I think the reason for the duplication is coming from the fact that you originally implemented it as part of the RCx architecture. As you originally did it. Could you after my changes harmonize that?